runtime/tr.source/trj9/optimizer/J9Simplifier.cpp

```diff
@@ -833,7 +833,7 @@ J9::Simplifier::simplifyiOrPatterns(TR::Node *node)
    secondChild = node->getSecondChild();
    if (!disableIORByteSwap &&
        !TR::Compiler->target.cpu.isBigEndian() &&  // the bigEndian case needs more thought
-       comp()->cg()->getSupportsIbyteswap() && secondChild->getOpCodeValue() == TR::ior)
```

Comment:
hmm, the bigEndian case needs more thought - wasn't was wrong the big endian case?

Reply:
It seems the problem was that both endian cases were incorrect, but since the little endian use case had been running in testing for 8 months without signs of problems we assumed that the problem was big-endian specific and disabled that code path. This change effectively swaps the little endian handling to what would have been (part of) the big endian code path.